### DISCUSSION TITLE
How to count the num of rows in a dataset and display

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-birt/birt#1713] [Q&A]
How to count the num of rows in a dataset and display in dynamic text in birt

**Comment by speckyspooky** [comment]
You have different options:
- Option 1: you can use "__rownum" on a table and show the maximum result. Add +1 to get the total count of rows because __rownum starts with 0.
- Option 2: you can use a parameter or variable to count the created rows. On table level you can add a small JS-script on a row at the method onCreate() and count the created rows.
- Option 3: you can use a parameter again and count the rows on dataSet-level and add the count on th method "onFetch"
- Option 4: you can count the rows on SQL level as a separate column and work with this special column

I hope there are some ideas for you which can help you.
